File: javascript/7_million_primes.js


function isPrime(numb){
  for(var i = 2; i <= Math.sqrt(numb); i++){
    if (numb % i == 0){
      return false;
    }
  }
  return true;
}


var count = 0
var numb = 2

while (true){
  if (isPrime(numb)){
    count += 1;
  }
  if (count == 10001){
    console.log(numb);
    break;
  }
  numb += 1;
}

File: python/7_million_primes.py
import os
import math

def isPrime(numb):
  for i in xrange(2,int(math.sqrt(numb))+1):
    if (numb % i == 0):
      return False
  return True


count = 0
numb = 2

while True:
  if isPrime(numb):
    count += 1
  if count == 10001:
    print numb
    break
  numb += 1
